### PR TITLE
Make BlockStateHolder extend Pattern.

### PR DESCRIPTION
**Pull request by wizjany**
Removes the need for wrapping them in BlockPattern.

~~todo: mark editsession methods as deprecated, probably?~~

### CODE REVIEW

File: worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockState.java [hunk 1]
```diff
@@ -207,6 +208,11 @@ public BaseBlock toBaseBlock() {
         return this.emptyBaseBlock;
     }
 
+    @Override
+    public BaseBlock apply(BlockVector3 position) {
```
Comment (kenzierocks) [inline worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockState.java]: This override isn't necessary. The overhead of `toBaseBlock()` is negligible.

File: worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BaseBlock.java [hunk 2]
```diff
@@ -166,6 +167,11 @@ public BaseBlock toBaseBlock() {
         return this;
     }
 
+    @Override
+    public BaseBlock apply(BlockVector3 position) {
```
Comment (kenzierocks) [inline worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BaseBlock.java]: Same here, the default in BSH is fine.

**Comment by MattBDev** [comment]
I know FAWE already does this but what prompted this change in WE?

**Comment by me4502** [comment]
I wonder if it's worth making BlockType a pattern and getting rid of BlockTypePattern too :thinking:

**Comment by wizjany** [comment]
Was actually considering making type return its default state as a pattern. FuzzyBlockState could be a RandomPattern by itself, though I'm not sure that's really useful. BlockTypePattern isn't a thing tho :P

**Comment by me4502** [comment]
Oh I must've deleted BlockTypePattern then oof... That's fair :joy:

But yeah it may make sense for it to return its default state as a pattern, just to make the code neater

**Comment by wizjany** [comment]
Yea there's a lot of getDefaultState going on. One of the issues with BlockType is that if people grab BlockType from BlockTypes, there's a potential for it to be null, which means we need to start null-checking pattern inputs. This hopefully shouldn't be too common tho, but for example `BlockTypes.SIGN` was null before the 1.14 block types update.

**Comment by me4502** [comment]
We could just do a checkNotNull, realistically that's an error on their part if they've grabbed a block that doesn't exist from BlockTypes (Or it's being used on the wrong version)

**Pull request merged**